catlearn/data/utils.py

```diff
@@ -0,0 +1,164 @@
+    for e_val in set(entities_val):
+        if not e_val in entities_set:
+            warning_str = f'Entity ${e_val} present in validation set is not present in training set. It was added.'
+            warnings.warn(str_color('W', warning_str), UserWarning)
+            logging.warning(warning_str)
+            entities_set.add(e_val)
+    for r_val in set(relations_val):
+        if not r_val in relations_set:
+            warning_str = f'Relation ${r_val} present in validation set is not present in training set. It was added.'
+            warnings.warn(str_color('W', warning_str), UserWarning)
+            logging.warning(warning_str)
+            relations_set.add(r_val)
+    # Check testing set on entities and relations not present in the training set
+    test: List[str] = read_file(os.path.join(path, raw_dataset_pat['test']))
+    entities_test, relations_test = get_entities_relations(test)
+    for e_test in set(entities_test):
+        if not e_test in entities_set:
+            warning_str = f'Entity ${e_test} present in testing set is not present in training set. It was added.'
+            warnings.warn(str_color('W', warning_str), UserWarning)
+            logging.warning(warning_str)
+            entities_set.add(e_test)
+    for r_test in set(relations_test):
+        if not r_test in relations_set:
+            warning_str = f'Relation ${r_test} present in testing set is not present in training set. It was added.'
+            warnings.warn(str_color('W', warning_str), UserWarning)
+            logging.warning(warning_str)
+            relations_set.add(r_test)
+    # Create dictionary with index
+    entity2id: Dict = {e: id for e, id in zip(entities_set, range(len(entities_set)))}
+    # id2entity = {id: e for e, id in entity2id.items()}
+    relation2id: Dict = {r: id for r, id in zip(relations_set, range(len(relations_set)))}
+    word2id: set = set([w.split('.')[0] for w in entities_set])
+    write_file(os.path.join(path, preproc_pat['wordset']), word2id)
+    # word2id: Dict = {k, id for k, id in zip(entities_set, range(len()))}
+    write_file(os.path.join(path, preproc_pat['relation2id']), relation2id)
+    write_file(os.path.join(path, preproc_pat['entity2id']), entity2id)
+
+
+def preprocess_word2vec(path_word2vec: str, path_dir_wordset: str):
+    """Create synthetic word2vec set with vectors for given entities
+
+    Args:
+        path_word2vec (str): path to word2vec model file location.
+        Either .txt or .pkl
+        path_entity_dic (str): path to the directory with entity2id file location
+        File name 
+    """
+    
+    word_set: List[str] = read_file(os.path.join(path_dir_wordset, preproc_pat['wordset']))
+    import word2vec
+    import torch
+    w2v_model = word2vec.load(path_word2vec)
+
+    w2v_short: Dict = {}
+    vec_dim = len(w2v_model['the'])
+    for k in word_set:
+        try: w2v_short[k] = w2v_model[k]
+        except:
```

Comment:
Never used raw `expect:` as it also catches synrax errors! At least go for `except Exception`

Reply:
As Maxime points out raw except catches syntax errors.
Also the expected exception is clear here (KeyError)

And this seems to me that this try, except is used for control flow, which is bad practice. use if/else statements.